proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,523 @@
+for customer contributions should satisfy the following goals:
+
+1. Users should know what is available for contribution. It should be clear
+   where they can expend effort that will be accepted.
+1. Maintainers should prioritize issues based on feedback provided by the
+   community to focus the efforts of the maintainers and the community. Users
+   should know how to provide that feedback.
+1. Users should know how to contribute. The should be able to reference
+   documentation to make high quality contributions. It should be accessible to
+   first time contributors.
+1. All requests should get a response from the maintainers. Communication from
+   maintainers should happen in a timely manner, even (especially!) if that’s
+   saying no. An initial response to open PRs should occur within [TBD] business
+   days.
+1. A contribution should not be left abandoned. The maintainers are responsible
+   see it through to a conclusion.
+1. Contributors should know exactly what to expect when interacting with the
+   maintainers. The current status should be clearly indicated and the next
+   steps well defined.
+
+## Specification
+
+ ![Figure A: Flowchart of contribution guide review
+process.](contrib-guide-high-level.png "Figure A: Flowchart of contribution
+guide review process.")
+
+**Figure A: Flowchart of contribution guide review process.**
+
+We propose to formalize the existing intake, implementation, and review stages
+to assure they are applied systematically and transparently. In addition, we
+will augment the current process with new stages to clarify what issues are
+available for contribution and what stage they are in. Figure A demonstrates how
+a request and contribution will proceed through the improved process.
+
+We will also publish an improved contributing guide to document the review
+process, acceptance criteria, and expectations for communication. We will use
+existing issue labeling in conjunction with a GitHub Project to provide
+transparent status to the community to indicate what is available for
+contribution and to track the progress of a contribution.
+
+The following sections detail the proposed improvements to the existing process.
+
+### Intake
+
+The intake process will be modified to explicitly identifies changes, through
+GitHub issues, where community contributions are welcome. The maintainers will
+actively review GitHub issues to make this determination. Depending on the issue
+type (bug or feature request) and area (code or documentation), a set of
+criteria must be met before marking it as ready for contribution. The criteria
+used to make that decision may include:
+
+1. The change is feasible for a community member to contribute. See [the
+   appendix](#changes-not-available-for-contribution) for a definition of what
+   would not be feasible for community contribution.
+1. There is enough detail provided for the maintainers to agree with the
```

Comment:
We should flesh out what "enough" is here by adding a follow up sentence. Maybe something like:

Any design discussions (e.g. public interface design, edge cases to consider, backwards compatibility concerns) must be resolved prior to submitting the pull request so that feedback during the pull request review revolves primarily around the implementation of the agreed upon design.